src/renderer/viz/expressions/ramp.js

```diff
@@ -180,39 +196,76 @@ export default class Ramp extends BaseExpression {
 
         if (this.palette.type === paletteTypes.IMAGE) {
             const images = this.palette._applyToShaderSource(getGLSLforProperty);
-
             return {
                 preface: input.preface + images.preface,
                 inline: `${images.inline}(imageUV, ${input.inline})`
             };
         }
 
-        let inline = `texture2D(texRamp${this._uid}, vec2((${input.inline}-keyMin${this._uid})/keyWidth${this._uid}, 0.5))`;
+        let inline = '';
+        let preface = `
+        uniform float keyMin${this._uid};
+        uniform float keyWidth${this._uid};
+        `;
+
+        let inputGLSL = `((${input.inline}-keyMin${this._uid})/keyWidth${this._uid})`;
         if (this.input.type === 'category' && this.input.isA(Property)) {
```

Comment:
Same here, I'll go for some extract methods, such as `_applyToShaderSourceImage`, `_applyToShaderSourceNumberArray`... and alike, to improve understanding for people without knowledge on shaders like me

Reply:
:+1: